cmd/snap-confine/mount-support-nvidia.c

```diff
@@ -583,8 +579,8 @@ void sc_mount_nvidia_driver(const char *rootfs_dir, const char *base_snap_name)
 	}
 
 	sc_identity old = sc_set_effective_identity(sc_root_group_identity());
-	int res = mkdir(SC_LIB, 0755);
-	if (res != 0 && errno != EEXIST) {
+	int res = sc_nonfatal_mkpath(SC_LIB, 0755);
```

Comment:
if the parent of /var/lib/snapd/lib does not exist then there's lot more other problems already, hence the original code created the leaf path only, but overall the change is ok